spring-transaction/src/test/kotlin/org/jetbrains/exposed/spring/TransactionSynchronizationTest.kt

```diff
@@ -0,0 +1,96 @@
+package org.jetbrains.exposed.spring
+
+import junit.framework.TestCase.assertTrue
+import org.junit.Before
+import org.springframework.transaction.TransactionDefinition
+import org.springframework.transaction.support.TransactionSynchronization
+import org.springframework.transaction.support.TransactionSynchronizationManager
+import kotlin.test.Test
+
+class TransactionSynchronizationTest : SpringTransactionTestBase() {
```

Comment:
This behaviour is somewhat tricky. The full `TransactionSynchronization` behaviour is actually not supported right now - because we handle savepoints outside of Springs knowledge, which means that the savepoint callbacks specifically do not trigger.

While that might be a somewhat incomplete support - I personally feel that covering 95% of the common use cases is better than not covering them, as listening to savepoint callbacks should be a quite niche thing.